app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.kt

```diff
@@ -0,0 +1,269 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.GestureDetector
+import android.view.GestureDetector.SimpleOnGestureListener
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import android.widget.HorizontalScrollView
+import android.widget.LinearLayout
+import androidx.annotation.IntRange
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.children
+import androidx.core.view.get
+import info.metadude.android.eventfahrplan.commons.logging.Logging
+import nerd.tuxmobil.fahrplan.congress.R
+import kotlin.math.abs
+import kotlin.math.ceil
+import kotlin.math.floor
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+
+class HorizontalSnapScrollView(
+    context: Context,
+    attrs: AttributeSet
+) : HorizontalScrollView(context, attrs) {
+
+    companion object {
+        private const val LOG_TAG = "HorizontalScrollView"
+        private const val NOT_INITIALIZED = Int.MIN_VALUE
+        private const val SWIPE_MIN_DISTANCE = 5
+        private const val SWIPE_THRESHOLD_VELOCITY = 2800
+        private const val MIN_PORTRAIT_SCROLLING_PERCENTAGE = 0.25
+
+        /**
+         * Calculates the number of columns to display at a time based on the physical dimensions and
+         * the screen density of the device and the column count of the schedule. Further limiting
+         * factors are a maximum column count to be displayed and a minimum column width.
+         */
+        @VisibleForTesting
+        fun calculateDisplayColumnCount(
+            availablePixels: Int,
+            columnsCount: Int,
+            maxColumnCountForLayout: Int,
+            densityScaleFactor: Float,
+            minColumnWidthDip: Int
+        ): Int {
+            val columnCountLimit = min(columnsCount, maxColumnCountForLayout)
+            if (columnCountLimit == 1) {
+                return 1
+            }
+            val availableDips = availablePixels.toFloat() / densityScaleFactor
+            val minWidthColumnCount = floor((availableDips / minColumnWidthDip).toDouble()).toInt()
+            val columnCount = min(minWidthColumnCount, columnCountLimit)
+            return max(1, columnCount)
+        }
+    }
+
+    private val gestureDetector: GestureDetector
+    private val logging: Logging = Logging.get()
+    private var xStart = 0
+    private var roomsCount = NOT_INITIALIZED
+    private var roomNames: HorizontalScrollView? = null
+    private var displayColumnCount: Int
+    private var columnWidth: Int
```

Comment:
That's probably not possible, as that getter is only used outside the class, but there are some places inside the class where the attribute is used directly and which would break. ([here](https://github.com/EventFahrplan/EventFahrplan/blob/c3b6203746b901bfa52bea3d24a6b9c99827089d/app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollView.java#L121) for example)

Reply:
I think the internal and external read usage should not be different. Both rely on that the property is initialized.

You can pull up the initialization from the `init { }` block.

``` kotlin
var columnWidth: Int = 0
    get() {
        check(roomsCount != NOT_INITIALIZED) {
            """The "roomsCount" field must be initialized before invoking "getColumnWidth"."""
        }
        return field
    }
    private set
```

Not tested!

Reply:
I thought so too, but it seems like the `onSizeChanged` method is called before the `roomCount` is initialized [here](https://github.com/EventFahrplan/EventFahrplan/blob/c3b6203746b901bfa52bea3d24a6b9c99827089d/app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.kt#L219). This should probably be looked into, but it seems that this is an issue that is a bit out of the scope of just transpiling.

Reply:
Interesting. Thank you for the clarification. I agree - this is out of scope.